src/node.js

```diff
@@ -390,6 +391,78 @@
     }
   };
 
+  startup.processPromises = function() {
+    // this should be kept in sync with PromiseRejectEvent in v8.h
+    var kPromiseRejectWithNoHandler = 0;
+    var kPromiseHandlerAddedAfterReject = 1;
+
+    var timers = NativeModule.require('timers');
+    var unhandledPromises = new WeakMap();
+    // A hack to work around uniterable WeakMaps.
+    var unhandledPromiseKeys = [];
```

Comment:
```js
Promise.reject().catch(function(){})
```

should not cause any notification

```js
var p = Promise.reject();
process.nextTick(function() {
    p.catch(function() {});
})
```

Should not cause any notification

```js
Promise.reject()
```

Should cause the notification `"unhandledRejection"`

```js
var p = Promise.reject();

setTimeout(function() {
    p.catch(function() {});
}, 1);
```

Should cause the notifications `"unhandledRejection"` and `"rejectionHandled"